src/lib/storage/index/adaptive_radix_tree/adaptive_radix_tree_index.cpp

```diff
@@ -32,50 +34,75 @@ AdaptiveRadixTreeIndex::AdaptiveRadixTreeIndex(const std::vector<std::shared_ptr
   std::vector<std::pair<BinaryComparable, ChunkOffset>> pairs_to_insert;
   pairs_to_insert.reserve(_indexed_segment->attribute_vector()->size());
 
+  const auto null_value_id = _indexed_segment->unique_values_count();
+  _index_null_postings.reserve(_indexed_segment->attribute_vector()->size());
+
   resolve_compressed_vector_type(*_indexed_segment->attribute_vector(), [&](const auto& attribute_vector) {
     auto chunk_offset = ChunkOffset{0u};
     auto value_id_it = attribute_vector.cbegin();
     for (; value_id_it != attribute_vector.cend(); ++value_id_it, ++chunk_offset) {
-      pairs_to_insert.emplace_back(BinaryComparable(ValueID{*value_id_it}), chunk_offset);
+      if (*value_id_it == null_value_id) {
+        _index_null_postings.emplace_back(chunk_offset);
+      } else {
+        pairs_to_insert.emplace_back(BinaryComparable(ValueID{*value_id_it}), chunk_offset);
+      }
     }
   });
 
+  _index_null_postings.shrink_to_fit();
   _root = _bulk_insert(pairs_to_insert);
 }
 
-BaseIndex::Iterator AdaptiveRadixTreeIndex::_lower_bound(const std::vector<AllTypeVariant>& values) const {
-  assert(values.size() == 1);
+AbstractIndex::Iterator AdaptiveRadixTreeIndex::_lower_bound(const std::vector<AllTypeVariant>& values) const {
+  Assert((values.size() == 1), "Adaptive Radix Tree Index expects exactly one input value");
+
+  if (variant_is_null(values[0])) {
+    return null_cbegin();
```

Comment:
This is most likely not the expected behavior. In the STL, `lower_bound` "[r]eturns an iterator pointing to the first element in the range [first, last) that is not less than (i.e. greater or equal to) value, or last if no such element is found". As NULL compares false to everything, this definition would lead us to always return `[null_]cend()`.

Maybe it is the callers job to deal with this and that `AbstractIndex::(lower|upper)_bound` should enforce that no NULL values are passed.

However, if we want to have multi column indexes that can deal with `WHERE a = 1 AND b IS NULL`, this behavior would be difficult. Maybe we need to discuss this in more detail.